### PR TITLE
strings - homework

### PR DESCRIPTION
**Pull request by aniscenko**

### CODE REVIEW

File: exercises/01_trim_right/01_trim_right.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+func TrimRight(s string) string {
+	for s != "" && InSet(s[len(s)-1], trimSet) {
+		s = s[:len(s)-1]
+	}
+	return s
+}
+
+func main() {
+	fmt.Printf("%q\n", TrimRight("   \r\r\r\n\t    hello"))
```
Comment (yarcat) [inline exercises/01_trim_right/01_trim_right.go]: nit: You don't need this string in your `main`.

File: exercises/01_trim_right/01_trim_right_test.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+import "testing"
+
+func TestTrimRight(t *testing.T) {
+	for _, tc := range []struct {
+		name  string
+		input string
+		want  string
+	}{
+		{"empty", "", ""},
```
Comment (yarcat) [inline exercises/01_trim_right/01_trim_right_test.go:11]: I also recommend adding tests like this:
- `"a"` just a single character
- `" "` just a seingle whitespace
- `"a "` a single whitespace and a character
There are often bugs that could be found by these simple corner cases. Your implementation shouldn't fail for those though. It seems good.

File: exercises/02_check_palindrom/02_check_palindrom.go [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package main
+
+func IsPalindrom(s string) bool {
+	r := []rune(s)
+	i, j := 0, len(r)-1
```
Comment (yarcat) [inline exercises/02_check_palindrom/02_check_palindrom.go]: nit: Consider declaring those within the `for` loop:
```go
for i, j := 0, len(r)-1; i < j; i, j = i+1, j-1 {
  ...
}
```

File: exercises/02_check_palindrom/02_check_palindrom_test.go [hunk 4]
```diff
@@ -0,0 +1,27 @@
+package main
+
+import "testing"
+
+func TestTrimRight(t *testing.T) {
```
Comment (yarcat) [inline exercises/02_check_palindrom/02_check_palindrom_test.go]: I believe it should be `TestIsPalindrome`

File: exercises/02_check_palindrom/02_check_palindrom_test.go [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+	for _, tc := range []struct {
+		input string
+		want  bool
+	}{
+		{"", true},
+		{"abc", false},
```
Comment (yarcat) [inline exercises/02_check_palindrom/02_check_palindrom_test.go:11]: I recommend having odd and even amount of characters for each set

File: exercises/06_count_words/06_count_words_test.go [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+		{"single word", "abc", 1},
+		{"single word in spaces", "   abc   ", 1},
+		{"two words", "abc def", 2},
+		{"two words in spaces", "   abc   def   ", 2},
+		{"unicode", "привет, мир!", 2},
+		{"string with numbers", "234 66 4", 3},
+		{"mixed string", "11 abc ", 2},
+	} {
+		if got := countWords(tc.input); got != tc.want {
+			fmt.Printf("%s\n\tFAIL: got %v, want %v\n", tc.name, got, tc.want)
```
Comment (yarcat) [inline exercises/06_count_words/06_count_words_test.go]: Since this is a proper test, you should be using `t.Errorf` instead of `fmt.Printf`!

File: exercises/06_count_words/06_count_words_test.go [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+			pass = false
+		}
+	}
+	if pass {
```
Comment (yarcat) [inline exercises/06_count_words/06_count_words_test.go]: With Go unit-tests you don't need this check. Using `t.Errorf` would mark your test as failing

File: exercises/03_count_byte/03_count_byte_test.go [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+import "testing"
+
+func TestCountByte(t *testing.T) {
+	for _, tc := range []struct {
+		name   string
+		input  string
+		inputb byte
+		want   int
+	}{
+		{"empty", "", 97, 0},
```
Comment (yarcat) [inline exercises/03_count_byte/03_count_byte_test.go:12]: nit: I recommend avoid using ascii codes directly. For instance `'a'` instead of `97`. `'a'` is also an integral constant in Go, which is exacly `97`, but it's very clear what's that

File: exercises/03_count_byte/03_count_byte.go [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+)
+
+func CountByte(s string, b byte) int {
+	var count int
+	for i := range []byte(s) {
```
Comment (yarcat) [inline exercises/03_count_byte/03_count_byte.go:11]: nit: it's cleaner to use values rather than indices in this case:
```go
for _, c := range []byte(s) {
  if c == b { count++ }
}
```

File: exercises/03_count_rune/03_count_rune.go [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+)
+
+func CountRune(s string, r rune) int {
+	var count int
+	for i := range []rune(s) {
```
Comment (yarcat) [inline exercises/03_count_rune/03_count_rune.go:11]: nit:
- string version of for-range defaults to runes:
- it's cleaner to use values

```go
for _, c := range s {
  if c == r {  count++ }
}
```

File: exercises/09_dedup/09_dedup.go [hunk 11]
```diff
@@ -0,0 +1,29 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+)
+
+func Dedup(s string) string {
```
Comment (yarcat) [inline exercises/09_dedup/09_dedup.go:9]: Wouldn't work for "\x00". You would return an empty string
